packages/grafana-ui/src/components/Table/TableCell.tsx

```diff
@@ -31,11 +59,172 @@ export const TableCell: FC<Props> = ({ cell, tableStyles, onCellFilterAdded, col
     innerWidth -= tableStyles.lastChildExtraPadding;
   }
 
-  return cell.render('Cell', {
-    field,
-    tableStyles,
-    onCellFilterAdded,
-    cellProps,
-    innerWidth,
-  }) as React.ReactElement;
+  const { style, ...otherCellProps } = cellProps;
+  const { justifyContent, ...commonCellStyles } = style || {};
+
+  return (
+    <div className={cx(styles.container, !previewEnabled && styles.containerHover)} style={commonCellStyles}>
+      {
+        cell.render('Cell', {
+          field,
+          tableStyles,
+          onCellFilterAdded,
+          cellProps: {
+            style: { justifyContent },
+            ...otherCellProps,
+          },
+          innerWidth,
+        }) as React.ReactElement
+      }
+      {previewEnabled && (
+        <>
+          <div
+            className={cx(
+              'actions',
+              isRightAligned && 'actionsLeft',
+              css`
+                svg {
+                  color: ${iconColor};
+                }
```

Comment:
this is really slow to have in each cell render (cx is super slow), and so is css function to dynamically generate styles